app/controllers/subscribe.js

```diff
@@ -0,0 +1,104 @@
+import Controller from '@ember/controller';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { RDS_TWITTER, APPS } from '../constants/urls';
+import { TOAST_OPTIONS } from '../constants/toast-options';
+import { PHONE_REGEX } from '../constants/regex';
+export default class SubscribeController extends Controller {
+  @service login;
+  @tracked isFormOpen = false;
+  @tracked email = '';
+  @tracked phone = '';
+  @tracked userData = null;
+  @tracked isLoading = false;
+  @tracked showSubscriptionModal = false;
+
+  RDS_TWITTER = RDS_TWITTER;
+
+  constructor() {
+    super(...arguments);
+    this.userData = this.login.userData;
+  }
+
+  get isLoggedIn() {
+    return this.login.isLoggedIn;
+  }
+
+  get isSubscribed() {
+    return this.login.userData.isSubscribed;
+  }
+
+  get isSubmitDisabled() {
+    return !this.email || (this.phone && !PHONE_REGEX.test(this.phone));
+  }
+
+  @action
+  toggleFormModal() {
+    this.isFormOpen = !this.isFormOpen;
+  }
+
+  @action
+  toggleSubscriptionModal() {
+    this.showSubscriptionModal = !this.showSubscriptionModal;
+    if (!this.showSubscriptionModal) {
+      window.location.reload();
+    }
+  }
+
+  @action
+  updateEmail(event) {
+    this.email = event.target.value;
+  }
+
+  @action
+  updatePhone(event) {
+    const input = event.target.value;
+    this.phone = input.replace(/[^\d+]/g, '');
```

Comment:
If you have made a constants file, you can add this regex in there and then use it

Reply:
Removed the comments